src/lib.rs

```diff
@@ -1,6 +1,138 @@
-#[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {
+use std::io::Write;
+use std::net::SocketAddrV4;
+use std::str::FromStr;
+
+pub struct Finder{
+    name: String,
+    socket: Option<SocketAddrV4>,
+    beacon_receiving_socket: std::net::UdpSocket,
+    last_beacon_time: std::time::Instant
+}
+
+pub struct Beacon{
+    data: Vec<u8>
```

Comment:
Would be good to have some docs that describes what `data` is and how it's laid out